src/sentry/integrations/notion/integration.py

```diff
@@ -0,0 +1,246 @@
+                "properties": {"Name": {"title": [{"text": {"content": data["title"]}}]}},
+                "children": [
+                    {
+                        "object": "block",
+                        "type": "heading_2",
+                        "heading_2": {
+                            "text": [
+                                {
+                                    "type": "text",
+                                    "text": {
+                                        "content": "Sentry integration",
+                                    },
+                                }
+                            ]
+                        },
+                    },
+                    {"object": "block", "type": "paragraph", "paragraph": {"text": []}},
+                    {
+                        "object": "block",
+                        "type": "paragraph",
+                        "paragraph": {
+                            "text": [{"type": "text", "text": {"content": data["description"]}}]
+                        },
+                    },
+                ],
+            }
+
+            notion_page = notion_client.create_page(data=payload)
+        except Exception as e:
+            self.raise_error(e)
+
+        key = notion_page["url"].split("-")
+
+        return {
+            "key": key[-1],
+            "title": data["title"],
+            "description": data["description"],
+        }
+
+    def after_link_issue(self, external_issue, **kwargs):
+        data = kwargs["data"]
+        notion_client = self.get_client()
+        comment = data["comment"]
+
+        if comment:
+            try:
+                notion_client.append_block(
+                    page_id=external_issue.key,
+                    comment=comment,
+                )
+
+            except Exception as e:
+                self.raise_error(e)
+
+    def get_issue(self, issue_id, **kwargs):
+        notion_client = self.get_client()
+        notion_page = notion_client.get_page(issue_id)
+
+        key = notion_page["url"].split("-")
+
+        return {
+            "key": key[-1],
+            "title": generate_page_title(notion_page),
+            "description": "Workspace: %s" % self.model.metadata["workspace_name"],
+        }
+
+
+class NotionIntegrationProvider(IntegrationProvider):
+    key = "notion"
+    name = "Notion"
+    metadata = metadata
+
+    features = frozenset(
+        [
+            IntegrationFeatures.ISSUE_BASIC,
+        ]
+    )
+    integration_cls = NotionIntegration
+
+    setup_dialog_config = {"width": 600, "height": 900}
+
+    def get_pipeline_views(self):
+        return [NotionInstallationRedirect()]
+
+    def build_integration(self, state):
+        code = state["code"]
+        setup_url = absolute_uri("/extensions/notion/setup/")
+        authentication = NotionApiClient().authenticate(code=code, redirect_uri=setup_url)
+
+        return {
+            "external_id": authentication["workspace_name"],
```

Comment:
`authentication.get("workspace_name")` instead so if "workspace_name" does not exist, you get a `None` rather than a `KeyError`